nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-core/src/main/java/org/nuxeo/ecm/admin/operation/HotReloadStudioSnapshot.java

```diff
@@ -76,7 +76,15 @@
 
     public static final String ID = "Service.HotReloadStudioSnapshot";
 
-    protected static boolean updateInProgress = false;
+    protected static volatile boolean updateInProgress = false;
+
+    protected static synchronized boolean setInProgress(boolean inProgress) {
+        if (updateInProgress == inProgress) {
```

Comment:
Doesn't volatile handle this already ?

Reply:
handle what ? the synchronization ? the logic ?

Reply:
The logic. I think `volatile` already handles this if